app/src/main/java/pm/gnosis/heimdall/ui/safe/recover/extension/ReplaceExtensionQrActivity.kt

```diff
@@ -0,0 +1,185 @@
+package pm.gnosis.heimdall.ui.safe.recover.extension
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.Html
+import android.text.SpannableStringBuilder
+import androidx.lifecycle.*
+import com.squareup.moshi.Moshi
+import io.reactivex.rxkotlin.plusAssign
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.rx2.await
+import pm.gnosis.heimdall.BuildConfig
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.data.remote.models.push.PushServiceTemporaryAuthorization
+import pm.gnosis.heimdall.data.repositories.PushServiceRepository
+import pm.gnosis.heimdall.di.components.ViewComponent
+import pm.gnosis.heimdall.reporting.ScreenId
+import pm.gnosis.heimdall.ui.base.ViewModelActivity
+import pm.gnosis.heimdall.ui.qrscan.QRCodeScanActivity
+import pm.gnosis.heimdall.utils.handleQrCodeActivityResult
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.common.utils.shareExternalText
+import pm.gnosis.svalinn.common.utils.snackbar
+import pm.gnosis.svalinn.common.utils.toast
+import pm.gnosis.svalinn.common.utils.visible
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+import timber.log.Timber
+import javax.inject.Inject
+import kotlinx.android.synthetic.main.screen_replace_extension_qr.replace_extension_back_arrow as backArrow
+import kotlinx.android.synthetic.main.screen_replace_extension_qr.replace_extension_bottom_panel as bottomPanel
+import kotlinx.android.synthetic.main.screen_replace_extension_qr.replace_extension_coordinator as coordinator
+import kotlinx.android.synthetic.main.screen_replace_extension_qr.replace_extension_extension_link as extensionLink
+import kotlinx.android.synthetic.main.screen_replace_extension_qr.replace_extension_progress_bar as progressBar
+
+class ReplaceExtensionQrActivity : ViewModelActivity<ReplaceExtensionQrContract>() {
+
+    override fun layout() = R.layout.screen_replace_extension_qr
+
+    override fun inject(component: ViewComponent) = viewComponent().inject(this)
+
+    override fun screenId() = ScreenId.REPLACE_BROWSER_EXTENSION_QR
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val safeAddress = (intent.extras[EXTRA_SAFE_ADDRESS] as String)?.asEthereumAddress()
+        safeAddress?.let {
+            viewModel.setup(it)
+        } ?: finish()
```

Comment:
since you do not `return` here the observer method will still be called even without the viewModel being setup, is this expected?